manifests/object/notification.pp

```diff
@@ -123,7 +123,7 @@
   if !is_string($user_groups) { validate_array($user_groups) }
   if $times { validate_hash ($times )}
   if $command { validate_string ($command )}
-  if $interval { validate_integer ($interval )}
+  if $interval { validate_re($interval, '^\d+\.?\d*[d|h|m|s]?$')}
```

Comment:
Please use: `\d+(\.\d+)?[dhms]?`

Your syntax is not really correct, its either `(d|h|m|s)` *or* `[dhms]`